src/config-loader.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Config handling - here because of strange babel errors.
+ * By Mia
+ * @author mia-pi-git
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+// @ts-ignore no typedef file
+import * as defaults from '../config/config-example';
+
+export type Configuration = typeof defaults;
+
+export function load(invalidate = false): Configuration {
+	const configPath = path.resolve(__dirname, '../../config/config.js');
+	if (invalidate) delete require.cache[configPath];
+	let config = defaults;
+	try {
+		config = {...config, ...require(configPath)};
+	} catch (err: any) {
+		if (err.code !== 'MODULE_NOT_FOUND') throw err; // Should never happen
+
+		console.log("config.js doesn't exist - creating one with default settings...");
+		fs.writeFileSync(
+			configPath,
+			fs.readFileSync(path.resolve(__dirname, '../../config/config-example.js'))
+		);
+	}
+	// so it's loaded after actions is originally loaded (preventing a crash)
+	process.nextTick(() => {
```

Comment:
I don't fully understand what this is doing, but it doesn't seem like something good. I assume when you clean up config usage this will no longer be necessary.

Reply:
This puts custom actions from config into the main list of loginserver actions. It's still necessary since the actions file references config stuff. Not sure if we can clean that up easily.